ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/thread/ThreadActionData.java

```diff
@@ -242,7 +242,7 @@ public static void purgeActions (Thread t, boolean unregister)
 
 					if (act != null)
                     {
-						act.removeChildThread(ThreadUtil.getThreadId(t));
+						act.removeChildThread(ThreadUtil.removeThreadId(t));
                     }
```

Comment:
The call to act.removeChildThread ... occurs in the method
```java
public static void purgeActions (Thread t, boolean unregister)
```
If the passed in thread t has not be added to the map before then ThreadUtil.removeThreadId(t) will be null.